### ISSUE TITLE
BUG: Nekoray 4.1.1 Update Causes URL Test Failures and Configuration Issues

### ISSUE THREAD

**Issue by Turbine8845** [Mahdi-zarei/nekoray#104]
## Describe the problem
After updating to version 4.1.1, I encountered several issues. Notably, the URL test feature no longer provides accurate ping results, and several configurations from various subscription links fail to work. Attempting a URL test on some sub groups results in errors, as seen in the log:

```log
sing-box: 1.10.0-alpha.29
2024/08/15 07:39:44 nekobox_core grpc server listening at 127.0.0.1:56814
grpc-status error code: 2 , error msg: create service: parse outbound[115]: unknown method: chacha20-poly1305
[Error] gRPC: QNetworkReply::NetworkError code: 301
Speedtest finished!
``` 
These issues persist even after performing a clean installation from the zip file provided in the Releases section. I use the following URL for latency tests: `http://www.google.com/generate_204`

Additionally, version 4.1.1 has removed several useful features such as TCP ping, Full test, Resolve Domain, and the ability to perform download speed tests using URLs like `http://cachefly.cachefly.net/10mb.test`.

It appears that others are also experiencing similar issues. Some Iranian and Chinese users have reported that their V2Ray configs no longer work with version 4.1.1. For reference, see the following issues:

- https://github.com/Mahdi-zarei/nekoray/issues/105

- https://github.com/Mahdi-zarei/nekoray/issues/103

- https://github.com/Mahdi-zarei/nekoray/issues/102

## Expected behavior:

- URL tests should provide accurate results without errors.

- Essential features such as TCP ping and download speed tests should be retained.

## Actual behavior:

- URL tests return incorrect results and numerous error messages.

- Many configurations that worked previously are now broken.

- Several useful features have been removed.

## How to reproduce
The following subscription links are from two public GitHub projects that aggregate V2Ray configurations from Telegram channels:

- https://raw.githubusercontent.com/yebekhe/TVC/main/subscriptions/xray/base64/mix

- https://raw.githubusercontent.com/itsyebekhe/HiN-VPN/main/subscription/normal/mix

To reproduce the issue, create a group using one of the above subscription links and perform a URL test. The log will display several error messages.

## log
```log
sing-box: 1.10.0-alpha.29
2024/08/15 07:39:44 nekobox_core grpc server listening at 127.0.0.1:56814
grpc-status error code: 2 , error msg: create service: parse outbound[115]: unknown method: chacha20-poly1305
[Error] gRPC: QNetworkReply::NetworkError code: 301
Speedtest finished!
grpc-status error code: 2 , error msg: create service: parse outbound[174]: unknown method: chacha20-poly1305
[Error] gRPC: QNetworkReply::NetworkError code: 301
Speedtest finished!
```
## Suggestion
I recommend marking version 4.1.1 as a beta release, as updating to it caused significant issues with my installation of Nekoray. There are still numerous problems that require further testing.

**Comment by Mahdi-zarei** [comment]
The part about some configurations not working anymore is really odd here, no changes were made to how configurations are started, please provide one of such configurations for further investigation. Regarding URL test issues, it will be solved in the next release, currently groups that contain configuration that are not valid will fail altogether, this will be fixed. About TCPing and speedtest, Maybe TCPing will return, but speedtest should be done properly by the user using tools designed for speedtesting, as the speedtest that was removed was not accurate enough.
The #103 issue and you both mention configurations that do not work in the new version, please first update to 4.1.1 and test again, and if it persists, provide information on the configuration so that the problem can be found, the "my config does not work anymore" statement is not making any point here.

**Comment by Turbine8845** [comment]
Thank you for the quick response. I encountered all the issues I mentioned in my comment while using version 4.1.1.

I understand your point about the speed test feature. While I agree that the built-in speed test wasn't entirely accurate, it did provide a rough indication of which configurations were functional, which was helpful when managing a large number of configs. Sometimes, performing a single latency test isn’t enough to identify a working V2Ray configuration. Given that manual download speed tests for each configuration are quite tedious, I would appreciate the return of this feature.

Regarding the configurations that fail to work, I haven’t identified a clear pattern yet, but I’ve noticed that some configurations from my subscription groups randomly stop working. I’ll continue to monitor this and will provide more detailed information as I gather it.

Currently, the most critical issue for me is the inability to perform bulk latency tests for the configurations in most of the subscription groups I use. This has significantly impacted my ability to manage and test configurations efficiently.

I’ll keep you updated with any new findings or patterns I observe.

**Comment by Turbine8845** [comment]
Here is one problematic config from one of my subscription links:

`
{"dns":{"rules":[{"domain":["series-a1.samanehha.co"],"domain_keyword":[],"domain_regex":[],"domain_suffix":[],"rule_set":[],"server":"dns-direct"}],"servers":[{"address":"https://8.8.8.8/dns-query","address_resolver":"dns-local","detour":"proxy","strategy":"","tag":"dns-remote"},{"address":"local","address_resolver":"dns-local","detour":"direct","strategy":"","tag":"dns-direct"},{"address":"rcode://success","tag":"dns-block"},{"address":"local","detour":"direct","tag":"dns-local"}]},"inbounds":[{"domain_strategy":"","listen":"127.0.0.1","listen_port":2080,"sniff":true,"sniff_override_destination":false,"tag":"mixed-in","type":"mixed"}],"log":{"level":"info"},"outbounds":[{"domain_strategy":"","method":"chacha20-poly1305","password":"W74XFALLLuw6m5IA","server":"series-a1.samanehha.co","server_port":443,"tag":"proxy","type":"shadowsocks","udp_over_tcp":false},{"tag":"direct","type":"direct"},{"tag":"block","type":"block"},{"tag":"dns-out","type":"dns"}],"route":{"final":"proxy","rule_set":[],"rules":[{"outbound":"dns-out","protocol":"dns"}]}}
`

When I try to test the following config I see the following messages in the log section:

```log
grpc-status error code: 2 , error msg: create service: parse outbound[3]: unknown method: chacha20-poly1305
[Error] gRPC: QNetworkReply::NetworkError code: 301
Speedtest finished!
```
When I try to connect to the mentioned config I see the following error message:

![Nekoray_problem](https://github.com/user-attachments/assets/a402b2c0-1ea6-4c1d-9f35-8812fb5decf3)

I also see the following messages in the log section when I try to connect to the config that I mentined:
```log
>>>>>>>> Starting profile [Shadowsocks] 🆔outline_vpn 🔒 SS-TCP-TLS 🆥🆥  109ms
<<<<<<<< Failed to start profile [Shadowsocks] 🆔outline_vpn 🔒 SS-TCP-TLS 🆥🆥  109ms
```

**Comment by Mahdi-zarei** [comment]
The config you have provided does not work on 4.0.10 either, and the error you see is coming directly from sing-box. Sing-box only supports the `ietf` version of `chacha20` algorithm, so practically there is no action I could take to resolve this issue on my end, as nekoray is a GUI for sing-box and if something is not supported by sing-box, nekoray cannot provide it either.

**Comment by Turbine8845** [comment]
> Here is one problematic config from one of my subscription links:
> 
> `{"dns":{"rules":[{"domain":["series-a1.samanehha.co"],"domain_keyword":[],"domain_regex":[],"domain_suffix":[],"rule_set":[],"server":"dns-direct"}],"servers":[{"address":"https://8.8.8.8/dns-query","address_resolver":"dns-local","detour":"proxy","strategy":"","tag":"dns-remote"},{"address":"local","address_resolver":"dns-local","detour":"direct","strategy":"","tag":"dns-direct"},{"address":"rcode://success","tag":"dns-block"},{"address":"local","detour":"direct","tag":"dns-local"}]},"inbounds":[{"domain_strategy":"","listen":"127.0.0.1","listen_port":2080,"sniff":true,"sniff_override_destination":false,"tag":"mixed-in","type":"mixed"}],"log":{"level":"info"},"outbounds":[{"domain_strategy":"","method":"chacha20-poly1305","password":"W74XFALLLuw6m5IA","server":"series-a1.samanehha.co","server_port":443,"tag":"proxy","type":"shadowsocks","udp_over_tcp":false},{"tag":"direct","type":"direct"},{"tag":"block","type":"block"},{"tag":"dns-out","type":"dns"}],"route":{"final":"proxy","rule_set":[],"rules":[{"outbound":"dns-out","protocol":"dns"}]}}`
> 
> When I try to URL test the following config I see the following messages in the log section:
> 
> ```
> grpc-status error code: 2 , error msg: create service: parse outbound[3]: unknown method: chacha20-poly1305
> [Error] gRPC: QNetworkReply::NetworkError code: 301
> Speedtest finished!
> ```
> 
> When I try to connect to the mentioned config I see the following error message:
> 
> ![Nekoray_problem](https://private-user-images.githubusercontent.com/166437392/358211975-a402b2c0-1ea6-4c1d-9f35-8812fb5decf3.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjQ4NTksIm5iZiI6MTcyMzcyNDU1OSwicGF0aCI6Ii8xNjY0MzczOTIvMzU4MjExOTc1LWE0MDJiMmMwLTFlYTYtNGMxZC05ZjM1LTg4MTJmYjVkZWNmMy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxNVQxMjIyMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNWU0ZmRjMWRkMTFiMTIzOWMyODVjNDI0NjYwNTk4YWE2MTJhNDA5ZDY0ZDkzOWI3ZDBhYjI2NzkyOWI5NDFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Cs62CHxXi2yVTvskCo2b7CZdxSUGmUtKpy8sngES1E4)
> 
> I also see the following messages in the log section when I try to connect to the config that I mentined:
> 
> ```
> >>>>>>>> Starting profile [Shadowsocks] 🆔outline_vpn 🔒 SS-TCP-TLS 🆥🆥  109ms
> <<<<<<<< Failed to start profile [Shadowsocks] 🆔outline_vpn 🔒 SS-TCP-TLS 🆥🆥  109ms
> ```

Another user also reported to have the URL test issue at https://github.com/Mahdi-zarei/nekoray/issues/107. I guess configs like the one I provided are responsible for all these new issues in version 4.1.1.

**Comment by Mahdi-zarei** [comment]
> Another user also reported to have the URL test issue at #107. I guess configs like the one I provided are responsible for all these new issues in version 4.1.1.

This, and those with splithttp transport. I will skip these kind of configs in the URLTest so that the rest of the group can be tested.

**Comment by Turbine8845** [comment]
> The config you have provided does not work on 4.0.10 either, and the error you see is coming directly from sing-box. Sing-box only supports the `ietf` version of `chacha20` algorithm, so practically there is no action I could take to resolve this issue on my end, as nekoray is a GUI for sing-box and if something is not supported by sing-box, nekoray cannot provide it either.

Thank you for your clarification. The problem in version 4.1.1 is that configs such as the one I provided are failing the URL test of the whole Group. In version 4.0.10 such configs doesn't fail the URL test of the whole group.

**Comment by Mahdi-zarei** [comment]
Try 4.1.2

**Comment by Turbine8845** [comment]
Thank you for the update. The problem is fixed in 4.1.2.
